ci/code_checks.sh

```diff
@@ -151,6 +151,14 @@ if [[ -z "$CHECK" || "$CHECK" == "patterns" ]]; then
     MSG='Check that no file in the repo contains tailing whitespaces' ; echo $MSG
     invgrep --exclude="*.svg" -RI "\s$" *
     RET=$(($RET + $?)) ; echo $MSG "DONE"
+
+    # Check that we use pytest.raises only as a context manager
+    #
+    # For any flake8-compliant code, the only way this regex gets
+    # matched is if there is no "with" statement preceding "pytest.raises"
+    MSG='Check for pytest.raises as context manager' ; echo $MSG
+    invgrep -R --include '*.py' -E '[[:space:]] pytest.raises' pandas/tests
+    RET=$(($RET + $?)) ; echo $MSG "DONE"
```

Comment:
> Small things, but I'd place this validation next to the assert_raises_regex, as they are related. 

Fair enough.

> Also, I'd remove the comment and just add a short note to MSG, IMO that would be even clearer, and more compact.

The comment should stay because it explains the logic of the regex used to check.  That being said, I can expand the `MSG` string.

Reply:
Expanded the `MSG` and moved it next to `assert_raises_regex` as requested.  I also added a new `MSG` to print in the CI to remind us about enabling the check in the future.